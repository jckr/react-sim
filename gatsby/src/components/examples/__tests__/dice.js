import React from 'react';
import renderer from 'react-test-renderer';
import { Die, DiceFrame, initDice, updateDice } from '../dice';
import Random from '../../helpers/random';

describe('Dice', () => {
  const params = {
    nbDice: 5,
  };
  let data;
  it('data function works', () => {
    const rng = new Random();
    data = initDice(params, rng.get);
    expect(data).toMatchSnapshot();
    const complete = jest.fn();
    for (let tick = 1; tick < 5; tick++) {
      data = updateDice({ data, tick, params, complete }, rng.get);
      expect(data).toMatchSnapshot();
    }
  });
  it('model renders properly', () => {
    for (let dots = 1; dots < 6; dots++) {
      expect(renderer.create(<Die value={dots} />).toJSON()).toMatchSnapshot();
    }
    const tree = renderer
      .create(
        <DiceFrame
          data={data}
          tick={5}
          params={params}
          theme={{ colors: { primary: 'red' } }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
