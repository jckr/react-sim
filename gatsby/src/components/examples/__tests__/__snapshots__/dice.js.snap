// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice data function works 1`] = `
Object {
  "average": 17.5,
  "rolls": Array [],
  "total": 0,
  "totals": Object {},
}
`;

exports[`Dice data function works 2`] = `
Object {
  "average": 0,
  "rolls": Array [
    6,
    2,
    6,
    5,
    6,
  ],
  "total": 25,
  "totals": Object {
    "25": 1,
  },
}
`;

exports[`Dice data function works 3`] = `
Object {
  "average": 12.5,
  "rolls": Array [
    6,
    4,
    2,
    5,
    1,
  ],
  "total": 18,
  "totals": Object {
    "18": 1,
    "25": 1,
  },
}
`;

exports[`Dice data function works 4`] = `
Object {
  "average": 14.333333333333334,
  "rolls": Array [
    2,
    5,
    1,
    6,
    6,
  ],
  "total": 20,
  "totals": Object {
    "18": 1,
    "20": 1,
    "25": 1,
  },
}
`;

exports[`Dice data function works 5`] = `
Object {
  "average": 15.75,
  "rolls": Array [
    1,
    1,
    5,
    3,
    4,
  ],
  "total": 14,
  "totals": Object {
    "14": 1,
    "18": 1,
    "20": 1,
    "25": 1,
  },
}
`;

exports[`Dice model renders properly 1`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "borderRadius": "3px",
      "height": 21,
      "marginRight": 10,
      "position": "relative",
      "width": 21,
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 8,
        "position": "absolute",
        "top": 8,
        "width": 3,
      }
    }
  />
</div>
`;

exports[`Dice model renders properly 2`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "borderRadius": "3px",
      "height": 21,
      "marginRight": 10,
      "position": "relative",
      "width": 21,
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 3,
        "position": "absolute",
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "position": "absolute",
        "right": 3,
        "width": 3,
      }
    }
  />
</div>
`;

exports[`Dice model renders properly 3`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "borderRadius": "3px",
      "height": 21,
      "marginRight": 10,
      "position": "relative",
      "width": 21,
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 3,
        "position": "absolute",
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 8,
        "position": "absolute",
        "top": 8,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "position": "absolute",
        "right": 3,
        "width": 3,
      }
    }
  />
</div>
`;

exports[`Dice model renders properly 4`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "borderRadius": "3px",
      "height": 21,
      "marginRight": 10,
      "position": "relative",
      "width": 21,
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 3,
        "position": "absolute",
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "position": "absolute",
        "right": 3,
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "left": 3,
        "position": "absolute",
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "position": "absolute",
        "right": 3,
        "width": 3,
      }
    }
  />
</div>
`;

exports[`Dice model renders properly 5`] = `
<div
  style={
    Object {
      "border": "1px solid #000",
      "borderRadius": "3px",
      "height": 21,
      "marginRight": 10,
      "position": "relative",
      "width": 21,
    }
  }
>
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 3,
        "position": "absolute",
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "position": "absolute",
        "right": 3,
        "top": 3,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "height": 3,
        "left": 8,
        "position": "absolute",
        "top": 8,
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "left": 3,
        "position": "absolute",
        "width": 3,
      }
    }
  />
  <div
    style={
      Object {
        "background": "#000",
        "borderRadius": "100%",
        "bottom": 3,
        "height": 3,
        "position": "absolute",
        "right": 3,
        "width": 3,
      }
    }
  />
</div>
`;

exports[`Dice model renders properly 6`] = `
<div
  className="css-m639qu"
>
  <div
    className="css-1kw0tju"
  >
    <div
      style={
        Object {
          "border": "1px solid #000",
          "borderRadius": "3px",
          "height": 21,
          "marginRight": 10,
          "position": "relative",
          "width": 21,
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 8,
            "position": "absolute",
            "top": 8,
            "width": 3,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "1px solid #000",
          "borderRadius": "3px",
          "height": 21,
          "marginRight": 10,
          "position": "relative",
          "width": 21,
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 8,
            "position": "absolute",
            "top": 8,
            "width": 3,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "1px solid #000",
          "borderRadius": "3px",
          "height": 21,
          "marginRight": 10,
          "position": "relative",
          "width": 21,
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "position": "absolute",
            "right": 3,
            "top": 3,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 8,
            "position": "absolute",
            "top": 8,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "bottom": 3,
            "height": 3,
            "left": 3,
            "position": "absolute",
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "bottom": 3,
            "height": 3,
            "position": "absolute",
            "right": 3,
            "width": 3,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "1px solid #000",
          "borderRadius": "3px",
          "height": 21,
          "marginRight": 10,
          "position": "relative",
          "width": 21,
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 8,
            "position": "absolute",
            "top": 8,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "bottom": 3,
            "height": 3,
            "position": "absolute",
            "right": 3,
            "width": 3,
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "border": "1px solid #000",
          "borderRadius": "3px",
          "height": 21,
          "marginRight": 10,
          "position": "relative",
          "width": 21,
        }
      }
    >
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "left": 3,
            "position": "absolute",
            "top": 3,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "height": 3,
            "position": "absolute",
            "right": 3,
            "top": 3,
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "bottom": 3,
            "height": 3,
            "left": 3,
            "position": "absolute",
            "width": 3,
          }
        }
      />
      <div
        style={
          Object {
            "background": "#000",
            "borderRadius": "100%",
            "bottom": 3,
            "height": 3,
            "position": "absolute",
            "right": 3,
            "width": 3,
          }
        }
      />
    </div>
  </div>
  <div
    className="css-u42cjd"
  >
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        5
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        6
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        7
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        8
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        9
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        10
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        11
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        12
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        13
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 50,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        14
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        15
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        16
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        17
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 50,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        18
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        19
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 50,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        20
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        21
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        22
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        23
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        24
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 50,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        25
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        26
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        27
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        28
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        29
      </div>
    </div>
    <div
      className="css-l3rx45"
    >
      <div
        className="css-180fwul"
      >
        <div
          style={
            Object {
              "backgroundColor": "red",
              "height": 0,
              "width": 10.76923076923077,
            }
          }
        />
      </div>
      <div
        className="css-1r49g1y"
      >
        30
      </div>
    </div>
  </div>
</div>
`;
